marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/reader/HubDataSourceReader.java

```diff
@@ -29,6 +33,11 @@
     private final Map<String, String> options;
     private final JsonNode initializationResponse;
     private final StructType sparkSchema;
+    private final HubClient hubClient;
```

Comment:
I don't think most of these should be class fields, which complicates the class. See next note below.